scripts/parser/open_ai_func.py

```diff
@@ -93,8 +98,57 @@ def get_user_permission(docs, folder_name):
     # Here we check for user permission before calling the API.
     user_input = input("Price Okay? (Y/N) \n").lower()
     if user_input == "y":
-        call_openai_api(docs, folder_name)
+        call_openai_api(docs, folder_name, use_s3, s3_bucket, s3_save_folder, s3_assume_role, aws_assume_role_profile)
     elif user_input == "":
-        call_openai_api(docs, folder_name)
+        call_openai_api(docs, folder_name, use_s3, s3_bucket, s3_save_folder, s3_assume_role, aws_assume_role_profile)
     else:
         print("The API was not called. No money was spent.")
+
+def upload_to_s3(s3_bucket, s3_save_folder, s3_assume_role, aws_assume_role_profile):
+    # Set defaults from environment variables if not provided via command line
+    s3_bucket = s3_bucket or os.environ.get("S3_BUCKET")
+    s3_save_folder = s3_save_folder or os.environ.get("S3_SAVE_FOLDER")
+    aws_assume_role_profile = aws_assume_role_profile or os.environ.get("AWS_ASSUME_ROLE_PROFILE")
+    
+    if not s3_bucket:
+        print("Error: S3_BUCKET environment variable is not set.")
```

Comment:
Please get acquainted with the concept of Errors and Exceptions in Python.

Reply:
Also the message is not completely correct in the context of the latest changes.